src/search/SearchMessagesScreen.js

```diff
@@ -98,13 +100,25 @@ class SearchMessagesScreen extends PureComponent<Props, State> {
       /* TODO: if an error makes it through the filter above,
          should we arrange to display something to the user? */
     }
-    // N.B.: `messages` is now set
 
     // A query is concluded. Report the message-list.
     this.setState({
-      messages,
+      messages: networkResponse.messages,
       isFetching: this.lastIdSent !== this.lastIdReceived,
     });
+
+    // store the fetched messages in the Redux store.
+    dispatch(
+      messageFetchComplete(
```

Comment:
One thing we've noticed about messageFetchComplete is that it takes a lot of arguments, so it can be tricky to remember what each one is for — it's not obvious from looking at this call site what, for example, the 20 passed as the fourth argument means.

So, it would be helpful, in a separate commit before this one, to change messageFetchComplete so that it has just one parameter, which is an object, where the properties are the names of the current parameters. The main advantage of doing it this way is that when you pass that argument, you're forced to label its properties at or near the call site, and people reading the code there can more easily understand what's happening. 🙂 

Concretely, messageFetchComplete should look something like this:

```
export const messageFetchComplete = (args: {
  messages: Message[],
  narrow: Narrow,
  anchor: number,
  numBefore: number,
  numAfter: number,
  foundNewest?: boolean,
  foundOldest?: boolean,
}): Action => {
  const { messages, narrow, anchor, numBefore, numAfter, foundNewest, foundOldest } = args;
  return {
    type: MESSAGE_FETCH_COMPLETE,
    messages,
    narrow,
    anchor,
    numBefore,
    numAfter,
    foundNewest,
    foundOldest,
  };
};
```

You'll also want to update all call sites of the messageFetchComplete action creator accordingly, in the same commit. (Just the action creator, not the handlers in fetchingReducer.js and narrowsReducer.js with the same name.)